pgaudit.c

```diff
@@ -655,7 +662,7 @@ log_audit_event(AuditEventStackItem *stackItem)
      * parameters if they have not already been logged for this substatement.
      */
     appendStringInfoCharMacro(&auditStr, ',');
-    if (!stackItem->auditEvent.statementLogged || !auditLogStatementOnce)
+    if (auditLogStatement && !(stackItem->auditEvent.statementLogged && auditLogStatementOnce))
```

Comment:
It's not clear to me why you changed the rest of the condition here, surely:
```
auditLogStatement && (!stackItem->auditEvent.statementLogged || !auditLogStatementOnce)
```
works just as well and is probably a bit more efficient?

Reply:
Don't remember ;-) --- but I think I find

    if (this and not (those-two))

easier to follow than

    if (this and (not that-one or not that-one))

I can change it back if you like.

I don't think efficiency is relevant here. The compiler will likely convert them back and forth depending on its opinion of the day.

Reply:
> The compiler will likely convert them back and forth depending on its opinion of the day.

Haha, true enough.  I agree that your way is more readable so let's stick with it.